File: packages/transform/src/features/http/HttpParameterTransformer.ts
import { HttpParameterProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace HttpParameterTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "http.parameter",
      write: HttpParameterProgrammer.write,
    });
}

File: packages/transform/src/features/http/HttpParameterTransformer.go
package http

import (
	"github.com/samchon/typia/packages/core/src/programmers"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var HttpParameterTransformer = httpParameterTransformerNamespace{}

type httpParameterTransformerNamespace struct{}

func (httpParameterTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	if programmers.ContainsType(call.TypeText, "number") {
		return "(input => Number(input))"
	}
	if programmers.ContainsType(call.TypeText, "boolean") {
		return `(input => input === "true")`
	}
	return "(input => input)"
}
